File: tourneyAPI/Models/Entities/Character.cs
namespace Entities;

using Enums;
using System.Diagnostics.CodeAnalysis;

[ExcludeFromCodeCoverage]
// Represents the chosen fighter attributes for a player.
public class Character
{
    public Guid Id { get; set; }
    public CharacterName characterName { get; set; }

    public Archetype archetype { get; set; }

    public FallSpeed fallSpeed { get; set; }

    public TierPlacement tierPlacement { get; set; }

    public WeightClass weightClass { get; set; }

}

File: tourneyClient/src/models/entities/Character.ts
import { Archetype } from "@/models/Enums/Archetype";
import { CharacterId } from "@/models/Enums/CharacterId";
import { CharacterName } from "@/models/Enums/CharacterName";
import { FallSpeed } from "@/models/Enums/FallSpeed";
import { TierPlacement } from "@/models/Enums/TierPlacement";
import { WeightClass } from "@/models/Enums/WeightClass";



export interface Character
{
    id: CharacterId;
    characterName: CharacterName;
    archetype: Archetype;
    fallSpeed: FallSpeed;
    tierPlacement: TierPlacement;
    weightClass: WeightClass;
}